backend/src/main/java/pl/edu/agh/backend/auth/AuthenticationService.java

```diff
@@ -0,0 +1,65 @@
+package pl.edu.agh.backend.auth;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.edu.agh.backend.auth.models.AuthenticationRequest;
+import pl.edu.agh.backend.auth.models.AuthenticationResponse;
+import pl.edu.agh.backend.auth.models.RegisterRequest;
+import pl.edu.agh.backend.config.JwtService;
+import pl.edu.agh.backend.db.models.Role;
+import pl.edu.agh.backend.db.models.User;
+import pl.edu.agh.backend.db.ports.UserRepository;
+import pl.edu.agh.backend.db.services.UserService;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtService jwtService;
+    private final AuthenticationManager authenticationManager;
+
+    public Optional<AuthenticationResponse> register(RegisterRequest request) {
+        var user = User.builder()
+                .username(request.getUsername())
+                .password(passwordEncoder.encode(request.getPassword()))
+                .role(Role.USER)
+                .build();
+        if (addUser(user).isPresent()) {
+            var jwtToken = jwtService.generateToken(user);
+            return Optional.of(AuthenticationResponse.builder()
+                    .token(jwtToken)
+                    .build());
+        }
+        else return Optional.empty();
+    }
+
+    public AuthenticationResponse authenticate(AuthenticationRequest request) {
```

Comment:
authenticate -> login 
bardziej intuicyjne

Reply:
to samo co wcześniej